include/exec/timed_scheduler.hpp

```diff
@@ -0,0 +1,84 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "../stdexec/execution.hpp"
+
+namespace exec {
+  namespace __schedule_after {
+    using namespace stdexec;
+    using std::chrono::duration;
+
+    struct schedule_after_t {
+      template <class _Scheduler, class _Rep, class _Period>
+        requires tag_invocable<schedule_after_t, _Scheduler, const duration<_Rep, _Period>&>
+              && sender<
+                   tag_invoke_result_t<schedule_after_t, _Scheduler, const duration<_Rep, _Period>&>>
+      auto operator()(_Scheduler&& __sched, const duration<_Rep, _Period>& __duration) const
+        noexcept(
+          nothrow_tag_invocable<schedule_after_t, _Scheduler, const duration<_Rep, _Period>&>)
+          -> tag_invoke_result_t<schedule_after_t, _Scheduler, const duration<_Rep, _Period>&> {
+        return tag_invoke(schedule_after_t{}, (_Scheduler&&) __sched, __duration);
+      }
+    };
+  }
+
+  using __schedule_after::schedule_after_t;
+  inline constexpr schedule_after_t schedule_after{};
+
+  namespace __schedule_at {
+    using namespace stdexec;
+    using std::chrono::time_point;
+
+    struct schedule_at_t {
+      template <class _Scheduler, class _Clock, class _Duration>
+        requires tag_invocable<schedule_at_t, _Scheduler, const time_point<_Clock, _Duration>&>
+              && sender<tag_invoke_result_t<
+                schedule_at_t,
+                _Scheduler,
+                const time_point<_Clock, _Duration>&>>
+      auto operator()(_Scheduler&& __sched, const time_point<_Clock, _Duration>& __time_point) const
+        noexcept(
+          nothrow_tag_invocable< schedule_at_t, _Scheduler, const time_point<_Clock, _Duration>&>)
+          -> tag_invoke_result_t< schedule_at_t, _Scheduler, const time_point<_Clock, _Duration>&> {
+        return tag_invoke(schedule_at_t{}, (_Scheduler&&) __sched, __time_point);
+      }
+    };
+  }
+
+  using __schedule_at::schedule_at_t;
+  inline constexpr schedule_at_t schedule_at{};
+
+  template <class _Scheduler, class _Duration>
+  concept __has_schedule_after = requires(_Scheduler&& __sched, const _Duration& __duration) {
+    { schedule_after((_Scheduler&&) __sched, __duration) } -> stdexec::sender;
+  };
+
+  template <class _Scheduler, class _TimePoint>
+  concept __has_schedule_at = requires(_Scheduler&& __sched, const _TimePoint& __time_point) {
+    { schedule_at((_Scheduler&&) __sched, __time_point) } -> stdexec::sender;
+  };
+
+  // TODO: Add more requirements such as __has_schedule_at or __has_now
+  template <class _Scheduler, class _Clock = std::chrono::system_clock>
+  concept timed_scheduler = /* std::chrono::is_clock_v<_Clock> && */
+    stdexec::scheduler<_Scheduler> && __has_schedule_after<_Scheduler, typename _Clock::duration>
+    && __has_schedule_at<_Scheduler, typename _Clock::time_point>;
```

Comment:
`timed_scheduler` needs a `now() -> time_point` cpo.

`now()` on the `timed_scheduler` allows a `virtual_time_scheduler` to be used in tests to do tests of seconds of elapsed time in ms.

virtual-time is also used in log processing and audio, video, and animations

Reply:
OK, I will add that cpo asap

Reply:
It's done now.